File: 2017/1/main.go
package main

import "github.com/loganbickmore/advent-of-code/utils"

func toIntArr(s string) []int {
	var arr = make([]int, len(s))
	for i, r := range s {
		arr[i] = utils.Atoi(string(r))
	}
	return arr
}

func part1(arr []int) int {
	var sum int
	list := append(arr, arr...)
	for i, n := range arr {
		if n == list[i+1] {
			sum += n
		}
	}
	return sum
}

func part2(arr []int) int {
	var sum int
	list := append(arr, arr...)
	for i, n := range arr {
		if n == list[i+len(arr)/2] {
			sum += n
		}
	}
	return sum
}

func main() {
	var input = toIntArr(utils.GetInput())
	println("Part1 ->", part1(input))
	println("Part2 ->", part2(input))
}

File: 2017/1/main.lua
for line in io.lines("input") do
  if line:len() >= 1 then
    input = f
    list = input .. input
  end
end

function solve(x)
  sum = 0
  i = 1
  for n in input:gmatch"." do
    if n == list:sub(i + x,i + x) then
      sum = sum + n
    end
    i = i + 1
  end
  return sum
end

print("part1 ->",solve(1))
print("part2 ->",solve(input:len() / 2))
